component/rules.jsonnet

```diff
@@ -8,85 +8,116 @@ local defaultAnnotations = {
 };
 
 
-local patch_node_filesystem_rules(rule) =
-  local node_fs_rules = std.set([
+local patchNodeFilesystemRules(rule) =
+  local nodeFsRules = std.set([
     'NodeFilesystemSpaceFillingUp',
     'NodeFilesystemAlmostOutOfSpace',
     'NodeFilesystemFilesFillingUp',
     'NodeFilesystemAlmostOutOfFiles',
   ]);
   rule {
     expr:
-      if std.setMember(rule.alert, node_fs_rules) then
+      if std.setMember(rule.alert, nodeFsRules) then
         'bottomk by (device, host_ip) (1, %s)' % rule.expr
       else
         rule.expr,
   };
 
+local patchPersistentVolumeRules(rule) =
+  local pvRules = std.set([
+    'KubePersistentVolumeUsageCritical',
+    'KubePersistentVolumeFullInFourDays',
+    'KubePersistentVolumeFillingUp',
+  ]);
+  rule {
+    expr:
+      if std.setMember(rule.alert, pvRules) then
+        // This will first deduplicate the available space per persistent volume  by taking the minimum grouped by pvc-name and namespace
+        // (Note: We explicitly use `min` and not `bottomk`. `bottomk` will produce duplicates if e.g. a RXO pod is recreated)
+        // The deduplicated metric is then matched with `kube_persistentvolumeclaim_info' which is always one. We
+        // do this by multiplying the two values.
+        // `*on(persistentvolumeclaim, namespace)` will mutliply the metrics and pvc_info that match on pvc-name and namespace
+        // `group_left(storageclass)` will add the storageclass label of the pvc_info to the resulting metric
+        // `label_replace((), "namespace", "$1", "exported_namespace", "(.+)")` renames the `exported_namespace` lable to
+        // `namespace` so that we are able to match the two metrics.
+        // Finally it will filter out shared storage classes if they are configured
+        (
+          'label_replace(min by (persistentvolumeclaim, exported_namespace) (%s), "namespace", "$1", "exported_namespace", "(.+)")'
```

Comment:
The `label_replace` should not be necessary anymore since we've fixed the broken labeling in the federation config in #19.